devel/ihaskell/Portfile

```diff
@@ -0,0 +1,119 @@
+long_description    \
+    IHaskell is a kernel for the Jupyter project, which allows you to \
+    use Haskell inside Jupyter frontends (including the console and \
+    notebook). For a tour of some IHaskell features, check out the \
+    demo Notebook at \
+    http://nbviewer.ipython.org/github/gibiansky/IHaskell/blob/master/notebooks/IHaskell.ipynb. More \
+    example notebooks are available on the wiki at \
+    https://github.com/gibiansky/IHaskell/wiki. The wiki also has more \
+    extensive documentation of IHaskell features.
+
+fetch.type          git
+
+# use these to specify python versions, python3 required 
+set python3_version 3.7
+set python3_version_nickname \
+    [join [lrange [split ${python3_version} .] 0 1] {}]
+
+# See https://github.com/gibiansky/IHaskell#mac,
+# https://github.com/gibiansky/IHaskell/blob/master/requirements.txt
+depends_build-append  \
+                    port:cabal \
+                    port:ghc \
+                    port:pango \
+                    port:python${python3_version_nickname} \
+                    port:py${python3_version_nickname}-cairo \
+                    port:py${python3_version_nickname}-ipykernel \
+                    port:py${python3_version_nickname}-ipywidgets \
+                    port:py${python3_version_nickname}-jupyter \
+                    port:py${python3_version_nickname}-jupyter_client \
+                    port:py${python3_version_nickname}-jupyter_core \
+                    port:py${python3_version_nickname}-magic \
+                    port:py${python3_version_nickname}-nbformat \
+                    port:py${python3_version_nickname}-pkgconfig \
+                    port:py${python3_version_nickname}-widgetsnbextension \
+                    port:zmq
+
+use_configure       no
+
+build {}
+
+# relative paths to ${prefix}
+set datadir         share/[string tolower ${name}]
+set jupyter_dir     share/jupyter
+
+# no jupyter_select yet, so hack PATH to find `which jupyter`: 
+# https://trac.macports.org/ticket/50608
+destroot.env \
+    PATH=$env(PATH):${prefix}/Library/Frameworks/Python.framework/Versions/${python3_version}/bin
+
+# destroot approach without destroot capability in cabal (or stack):
+# create symlink in ${prefix}, install using symlink, delete symlink
+destroot {
+    # create a symbolic link in ${prefix} that points to a sandbox
+    xinstall -m 0755 -d ${workpath}/${datadir}
+    delete ${prefix}/${datadir}
+    ln -s ${workpath}/${datadir} ${prefix}/${datadir}
+    # see https://github.com/gibiansky/IHaskell/issues/1061
+    # add /usr/lib to path to avoid iconv linking issue
+    # see http://blog.omega-prime.co.uk/2011/01/28/solving-ghc-iconv-problems-on-os-x-10-6/
+    xinstall -W ${worksrcpath} -m 0644 ${filespath}/cabal.config .
+    system -W ${worksrcpath} "/bin/sh <<BUILD_AND_DESTROOT_IHASKELL
+        cabal update
+        cabal install cpphs happy
+        cabal sandbox init
+        cabal sandbox add-source ./ipython-kernel
```

Comment:
You probably want `cabal sandbox add-source ./ghc-parser` here as well.

Reply:
Thanks. Giving this a go. By the way, is there some cabal-based reason not to issue this single command?
```
cabal sandbox add-source . ./ghc-parser ./ipython-kernel
```

Reply:
I haven't tried that, but if it works I don't see any reason why not.

Reply:
That fixed it. Thanks again!